src/rgw/rgw_tag.h

```diff
@@ -0,0 +1,39 @@
+#ifndef RGW_TAG_H
+#define RGW_TAG_H
+
+#include <map>
+#include <string>
+#include <include/types.h>
+
+#include "rgw_common.h"
+
+class RGWObjTags
+{
+ protected:
+  std::map <std::string, std::string> tags;
```

Comment:
It might be worth using a boost::container::flat_map. Since we're splitting into a vector we could sort it.